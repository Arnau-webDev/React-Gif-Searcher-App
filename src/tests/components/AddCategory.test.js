import React from 'react';
import '@testing-library/jest-dom';

const { shallow } = require('enzyme');
const { AddCategory } = require('../../components/AddCategory');

describe('Pruebas AddCategory', () => {
	const setCategories = jest.fn();
	let value;
	let wrapper;
	let categories = [];

	beforeEach(() => {
		jest.clearAllMocks();
		value = 'Hola Mundo';
		categories = ['Marvel', 'Hola'];
		wrapper = shallow(
			<AddCategory categories={categories} setCategories={setCategories} />
		);
	});

	test('Debe mostrar el snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('No debe de hacer la peticiÃ³n fetch con el onSubmit', () => {
		wrapper
			.find('form')
			.simulate('submit', { target: { value }, preventDefault: () => {} });

		expect(setCategories).not.toHaveBeenCalled();
	});

	test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
		wrapper.find('input').simulate('change', { target: { value } });
		wrapper.find('form').simulate('submit', { preventDefault: () => {} });

		expect(setCategories).toHaveBeenCalled();
		expect(setCategories).toHaveBeenCalledTimes(1);
		expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

		const inputText = wrapper.find('input').props().value;
		expect(inputText).toBe('');
	});
});
