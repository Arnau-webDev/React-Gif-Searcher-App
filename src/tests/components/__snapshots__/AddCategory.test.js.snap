// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas AddCategory Debe mostrar el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Type about something..."
    type="text"
    value=""
  />
</form>
`;
